From 0b145e1e52b2d806113d1c39b16d024d8abdc26b Mon Sep 17 00:00:00 2001
From: Shuhao Wu <shuhao@shuhaowu.com>
Date: Mon, 5 Aug 2019 22:24:42 -0400
Subject: [PATCH 3/4] Increased limit on loot tracker SELECT

Also immediately submit loot on drop instead of waiting for 5 min.
---
 .../http/service/loottracker/LootTrackerController.java       | 4 ++--
 .../client/plugins/loottracker/LootTrackerPlugin.java         | 3 +++
 2 files changed, 5 insertions(+), 2 deletions(-)

diff --git a/http-service/src/main/java/net/runelite/http/service/loottracker/LootTrackerController.java b/http-service/src/main/java/net/runelite/http/service/loottracker/LootTrackerController.java
index 9b810f672..b886151a7 100644
--- a/http-service/src/main/java/net/runelite/http/service/loottracker/LootTrackerController.java
+++ b/http-service/src/main/java/net/runelite/http/service/loottracker/LootTrackerController.java
@@ -82,7 +82,7 @@ public class LootTrackerController
 	}
 
 	@GetMapping
-	public Collection<LootAggregate> getLootAggregate(HttpServletRequest request, HttpServletResponse response, @RequestParam(value = "count", defaultValue = "1024") int count, @RequestParam(value = "start", defaultValue = "0") int start) throws IOException
+	public Collection<LootAggregate> getLootAggregate(HttpServletRequest request, HttpServletResponse response, @RequestParam(value = "count", defaultValue = "16384") int count, @RequestParam(value = "start", defaultValue = "0") int start) throws IOException
 	{
 		SessionEntry e = auth.handle(request, response);
 		if (e == null)
@@ -107,4 +107,4 @@ public class LootTrackerController
 
 		service.delete(e.getUser(), eventId);
 	}
-}
\ No newline at end of file
+}
diff --git a/runelite-client/src/main/java/net/runelite/client/plugins/loottracker/LootTrackerPlugin.java b/runelite-client/src/main/java/net/runelite/client/plugins/loottracker/LootTrackerPlugin.java
index de4f6c1ed..532651ed7 100644
--- a/runelite-client/src/main/java/net/runelite/client/plugins/loottracker/LootTrackerPlugin.java
+++ b/runelite-client/src/main/java/net/runelite/client/plugins/loottracker/LootTrackerPlugin.java
@@ -427,6 +427,9 @@ public class LootTrackerPlugin extends Plugin
 			{
 				queuedLoots.add(lootRecord);
 			}
+
+			// PATCHED: We can just immediately submit it because don't need the rate limiting locally
+			submitLoot();
 		}
 
 		eventBus.post(new LootReceived(name, combatLevel, type, items));
-- 
2.25.1

